src/main/java/org/wso2/carbon/esb/connector/auth/AmazonLambdaAuthConnector.java

```diff
@@ -0,0 +1,438 @@
+                if (StringUtils.isNotEmpty(tempParam)) {
+                    headersParamsMap.put(headerParametersMap.get(key),
+                            tempParam.replaceAll
+                                    (AmazonLambdaConstants.TRIM_SPACE_REGEX, AmazonLambdaConstants.EMPTY_STR));
+                }
+            }
+            final SortedSet<String> headerKeys = new TreeSet<>(headersParamsMap.keySet());
+            for (String key : headerKeys) {
+                String headerValues = headersParamsMap.get(key);
+                canonicalHeader.append(key.toLowerCase()).append(AmazonLambdaConstants.COLON)
+                        .append(headerValues).append(AmazonLambdaConstants.NEW_LINE);
+                signedHeader.append(key.toLowerCase());
+                signedHeader.append(AmazonLambdaConstants.SEMI_COLON);
+            }
+            //Appending canonicalHeader to canonicalRequest.
+            canonicalRequest.append(canonicalHeader.toString());
+            canonicalRequest.append(AmazonLambdaConstants.NEW_LINE);
+            // Removes unwanted semi-colon at the end of the signedHeader string
+            String signedHeaders = "";
+            if (signedHeader.length() > 0) {
+                signedHeaders = signedHeader.substring(0, signedHeader.length() - 1);
+            }
+            //Appending signedHeaders to canonicalRequest.
+            canonicalRequest.append(signedHeaders);
+            canonicalRequest.append(AmazonLambdaConstants.NEW_LINE);
+            //Payload Building from the payload parameter and value given by user.
+            String requestPayload = "";
+            final Map<String, String> payloadParametersMap = ParameterNamesMap.getPayloadParameterNamesMap();
+            final StringBuilder payloadBuilder = new StringBuilder();
+            for (Map.Entry<String, String> entry : payloadParametersMap.entrySet()) {
+                String key = entry.getKey();
+                String tempParam = payloadParametersValueMap.get(key);
+                if (StringUtils.isNotEmpty(tempParam)) {
+                    payloadParamsMap.put(payloadParametersMap.get(key),
+                            tempParam.replaceAll
+                                    (AmazonLambdaConstants.TRIM_SPACE_REGEX, AmazonLambdaConstants.EMPTY_STR));
+                }
+            }
+            final SortedSet<String> payloadKeys = new TreeSet<>(payloadParamsMap.keySet());
+            if (payloadKeys.equals(AmazonLambdaConstants.PAYLOAD)) {
+                payloadBuilder.append(payloadParamsMap.get(AmazonLambdaConstants.PAYLOAD));
+            } else {
+                for (String key : payloadKeys) {
+                    String payloadValues = payloadParamsMap.get(key);
+                    payloadBuilder.append('"');
+                    payloadBuilder.append(key);
+                    payloadBuilder.append('"');
+                    payloadBuilder.append(':');
```

Comment:
you can append like
 payloadBuilder.append('"').append(key).append('"').append(':');

Reply:
Fix with 408c7bd